### PR TITLE
Adding animation on opening a new activity

### PR DESCRIPTION
**Pull request by alexandragnj**
## New features

-Adding activity animation code

## Changes in code

-Call makeClipRevealAnimation() method from ActivityOptions

-Change the minimum SDK version

## Screenshot(GIF)
Apr-18-2022 09-06-07

### CODE REVIEW

File: app/src/main/java/com/example/geoquiz/QuizActivity.kt [hunk 1]
```diff
@@ -150,10 +154,17 @@ class QuizActivity : AppCompatActivity() {
...
             val answerIsTrue = quizViewModel.currentQuestionAnswer
             val intent = CheatActivity.newIntent(this@QuizActivity, answerIsTrue)
-            startActivityForResult(intent, REQUEST_CODE_CHEAT)
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                val options = ActivityOptions
+                    .makeClipRevealAnimation(view, 0, 0, view.width, view.height)
+                startActivityForResult(intent, REQUEST_CODE_CHEAT, options.toBundle())
+            } else {
+                startActivityForResult(intent, REQUEST_CODE_CHEAT)
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/QuizActivity.kt]: Please replace startActivityForResult with resultLauncher from the link I sent you

File: app/src/main/java/com/example/geoquiz/QuizActivity.kt [hunk 2]
```diff
@@ -150,10 +154,17 @@ class QuizActivity : AppCompatActivity() {
             ).show()
         }
 
-        cheatButton.setOnClickListener {
+        cheatButton.setOnClickListener { view ->
             val answerIsTrue = quizViewModel.currentQuestionAnswer
             val intent = CheatActivity.newIntent(this@QuizActivity, answerIsTrue)
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/QuizActivity.kt]: When things get crowded inside lambda brackets, that's a sign that we can move code to a function that will be called upon button press. I would move the code to a `openCheatActivity()` function

File: app/src/main/java/com/example/geoquiz/QuizActivity.kt [hunk 3]
```diff
@@ -108,6 +111,7 @@ class QuizActivity : AppCompatActivity() {
         cheatButton = findViewById(R.id.cheat)
     }
 
+    @SuppressLint("RestrictApi")
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/QuizActivity.kt]: What's the purpose of this lint suppressing?

File: app/build.gradle [hunk 4]
```diff
@@ -8,7 +8,7 @@ android {
 
     defaultConfig {
         applicationId "com.example.geoquiz"
-        minSdk 21
+        minSdk 23
```
Comment (iugaliviu) [inline app/build.gradle]: Why the change for minSdk?

Comment (alexandragnj) [reply app/build.gradle]: It was only for testing purpose

**Review by iugaliviu — APPROVED** [review]
Code looks good. Will approve. Still some files added to the repo, they should not be here. We will have to make sure for the next project that .gitignore is working properly

File: app/build.gradle [hunk 5]
```diff
@@ -2,6 +2,7 @@ plugins {
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
 }
+apply plugin: 'kotlin-android'
```
Comment (NegreaVlad) [inline app/build.gradle]: this is equivalent to what is on top with the plugins clause. 
you can add it there with id 'kotlin-android'

File: app/build.gradle [hunk 6]
```diff
@@ -33,13 +34,20 @@ android {
...
+    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:2.4.1"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 
     testImplementation 'junit:junit:4.13.2'
 
     androidTestImplementation 'androidx.test.ext:junit:1.1.3'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+}
+repositories {
```
Comment (NegreaVlad) [inline app/build.gradle]: Repositories are usually defined on top. Below plugins is a good place

File: app/src/main/java/com/example/geoquiz/QuizActivity.kt [hunk 7]
```diff
@@ -80,27 +68,19 @@ class QuizActivity : AppCompatActivity() {
...
-    }
-
-    override fun onDestroy() {
-        super.onDestroy()
-        Log.d(TAG, "onDestroy() called")
-    }
-
     private fun restoreState(savedInstanceState: Bundle?) {
-        if (savedInstanceState != null) {
+        savedInstanceState?.let {
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/QuizActivity.kt]: It's hard to follow why the if has been changed to let. If you do not benefit from having a non null value provided by let, I prefer to use the if not null clause. It's easier to understand because this way, I don't know where the `it` from line 76, comes from, I have to really look closely at the code to understand

Comment (NegreaVlad) [reply app/src/main/java/com/example/geoquiz/QuizActivity.kt]: If you want to use the let, name the param inside `let` to bundle or something and use it instead of `savedInstanceState `

File: app/src/main/java/com/example/geoquiz/QuizActivity.kt [hunk 8]
```diff
@@ -150,22 +130,40 @@ class QuizActivity : AppCompatActivity() {
...
     private fun updateQuestion() {
         val questionTextResId = quizViewModel.currentQuestionText
         questionTextView.setText(questionTextResId)
-        falseButton.isClickable = !(quizViewModel.questionWasAnswered[quizViewModel.currentIndex]!!)
-        trueButton.isClickable = !(quizViewModel.questionWasAnswered[quizViewModel.currentIndex]!!)
+        falseButton.isClickable = !(quizViewModel.questionWasAnswered[quizViewModel.currentIndex])
+        trueButton.isClickable = !(quizViewModel.questionWasAnswered[quizViewModel.currentIndex])
+    }
+
+    private fun setAnswerButtonsClickable() {
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/QuizActivity.kt]: Code doesn't match the name. It makes them unclickable

File: app/src/main/java/com/example/geoquiz/QuizViewModel.kt [hunk 9]
```diff
@@ -9,13 +9,13 @@ class QuizViewModel : ViewModel() {
     var correctAnswers: Int = 0
     var messageResId: Int = 0
 
-    var questionBank = listOf<Question>(
+    var questionBank = listOf(
         Question(R.string.question_australia, true),
         Question(R.string.question_oceans, true),
         Question(R.string.question_mideast, false),
         Question(R.string.question_africa, false),
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/QuizViewModel.kt]: Remove empty line and reformat code

**Review by NegreaVlad — COMMENTED** [review]
Some small comments. Looks good overall

**Review by NegreaVlad — APPROVED** [review]
lgtm
